biogrid/libs/biogrid-simulator/src/lib/bioenergy-source/solar-panel.ts

```diff
@@ -71,12 +71,12 @@ export class SolarPanel extends EnergySource {
    * This method returns the current power that can be generated by
    * the solar panel at that given time
    */
-  async getEnergyInJoules(date?: Date): Promise<Power> {
-    return await this.getPowerAmount(this.date || date);
+  async getEnergyInKilowattHour(date?: Date): Promise<Power> {
+    return await this.getPowerAmount(date || this.date);
   }
 
-  async isEmpty() {
-    return (await this.getEnergyInJoules()) === 0;
+  async isEmpty(date: Date) {
```

Comment:
nit: it seems a bit unfortunate that all these functions have to be marked as async because this.getPowerAmount calls await this.weatherLib.setup(); which is the only promise in this entire chain that needs to be called. Can we call this.weatherLib.setup in the constructor of this class instead, that way the rest of the nested functions don't have to be async?

Reply:
Ok sure, I can mark this for a future issue

Reply:
sg :)